[coap] use of `nullptr` in C API

I just fetched the most recent Zephyr zephyr-v2.5.0-148-g4ee876b797b1 which just merged #4247 (CoAP block transfers). 

Build fails... (board is nrf52)...both crosstools arm compiler and arm gcc.
```
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h: In function 'otCoapSendRequestBlockWise':
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h:946:108: error: 'nullptr' undeclared (first use in this function)
  946 |     return otCoapSendRequestBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, aHandler, aContext, nullptr,
      |                                                                                                            ^~~~~~~
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h:946:108: note: each undeclared identifier is reported only once for each function it appears in
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h: In function 'otCoapSendResponseBlockWise':
build/ZEPHYR/modules/lib/openthread/include/openthread/coap.h:1107:89: error: 'nullptr' undeclared (first use in this function)
 1107 |     return otCoapSendResponseBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, nullptr, aContext,
      |                                                                                         ^~~~~~~
```

--- a/include/openthread/coap.h
+++ b/include/openthread/coap.h
@@ -943,7 +943,8 @@ static inline otError otCoapSendRequestBlockWise(otInstance *                aIn
                                                  otCoapBlockwiseTransmitHook aTransmitHook,
                                                  otCoapBlockwiseReceiveHook  aReceiveHook)
 {
-    return otCoapSendRequestBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, aHandler, aContext, nullptr,
+    // NOLINTNEXTLINE(modernize-use-nullptr)
+    return otCoapSendRequestBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, aHandler, aContext, NULL,
                                                     aTransmitHook, aReceiveHook);
 }
 
@@ -1104,8 +1105,8 @@ static inline otError otCoapSendResponseBlockWise(otInstance *                aI
                                                   void *                      aContext,
                                                   otCoapBlockwiseTransmitHook aTransmitHook)
 {
-    return otCoapSendResponseBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, nullptr, aContext,
-                                                     aTransmitHook);
+    // NOLINTNEXTLINE(modernize-use-nullptr)
+    return otCoapSendResponseBlockWiseWithParameters(aInstance, aMessage, aMessageInfo, NULL, aContext, aTransmitHook);
 }
 
 /**

--- a/include/openthread/instance.h
+++ b/include/openthread/instance.h
@@ -53,7 +53,7 @@ extern "C" {
  * @note This number versions both OpenThread platform and user APIs.
  *
  */
-#define OPENTHREAD_API_VERSION (70)
+#define OPENTHREAD_API_VERSION (71)
 
 /**
  * @addtogroup api-instance

